### PR TITLE
Exclude JsxImport during finding references

### PR DESCRIPTION
**Pull request by armanio123**
Fixes https://github.com/microsoft/TypeScript/issues/53159#issuecomment-1459191205

Synthetic nodes cause an Assert exception when finding references. This PR excludes the jsx import when finding references as they are created as synthetize declaration.

### CODE REVIEW

File: src/services/importTracker.ts
Comment (sandersn) [inline src/services/importTracker.ts:479]: I think jsxImport can be lifted out of the the loop.

File: src/services/importTracker.ts
Comment (sandersn) [inline src/services/importTracker.ts:481]: this will skip even explicit, non-synthetic references to the jsx implicit import, right? Is there a way to check that the import declaration is synthetic, because those are the only ones we want/need to skip.

**Review by sandersn — CHANGES_REQUESTED** [review]
Needs a find-all-refs test. Those are in $ts/tests/cases/fourslash/findAllRef* -- I hope there are some existing JSX ones that you can copy to get the structure from.

File: src/services/importTracker.ts
Comment (armanio123) [reply src/services/importTracker.ts:479]: Fixed.

File: src/services/importTracker.ts
Comment (armanio123) [reply src/services/importTracker.ts:481]: That's correct. Fixed.

**Comment by DanielRosenwasser** [comment]
Can you discuss whether this or https://github.com/microsoft/TypeScript/pull/51319 is the right fix, and mark this PR as fixing https://github.com/microsoft/TypeScript/issues/51221 ?

File: src/services/importTracker.ts
Comment (weswigham) [inline src/services/importTracker.ts:482]: You don't need to specifically check for the `jsxImport`, any node for which `nodeIsSynthesized` is true is going to be a hassle for the import tracker, since it has no associated positions to report. Just checking that will also remove any errors related to finding all references on the synthetic `tslib` import we also add. However, just skipping the result leaves a bit lacking, no? That's why the other PR adds a new reference type and actually records the file as a reference for the import.

**Review by weswigham — CHANGES_REQUESTED** [review]
If you just wanna quickly fix the crash, just doing the `nodeIsSynthesized` check (and skipping all the jsx-specific-filtering stuff) should broadly fix the crash and any similar ones with `tslib` imports. The other PR goes farther, though, which may be more what we want. Given the other PR has approvals already, @DanielRosenwasser would you prefer that change? IIRC, it's good to merge provided you're fine with FAR on a `tslib` import landing you on the first JSX tag in a file if present (rather than, say, the first statement which actually uses a helper when transpiled (which I don't think is easily tracked)).

**Comment by DanielRosenwasser** [comment]
On the other PR, I was waiting for you to address Sheetal's feedback regarding the tslib changes. If you go to the first statement for those cases (or for all cases even) I think that's better than the crash.

**Comment by weswigham** [comment]
> If you go to the first statement for those cases (or for all cases even) I think that's better than the crash.

Alright, it was a bit unclear what the `tslib` reference should be, given where the references in the generated code actually end up. If we're fine with the first statement of the file, that's easy.

**Comment by armanio123** [comment]
Closing in favor of https://github.com/microsoft/TypeScript/pull/51319

**Pull request closed without merging**
